server/src/main/java/org/opensearch/script/ScoreScript.java

```diff
@@ -144,6 +154,25 @@ public Map<String, ScriptDocValues<?>> getDoc() {
         return leafLookup.doc();
     }
 
+    public Object getTermFrequency(TermFrequencyFunctionName functionName, String field, String val) throws IOException {
+        String cacheKey = (val == null)
+            ? String.format(Locale.ROOT, "%s-%s", functionName, field)
+            : String.format(Locale.ROOT, "%s-%s-%s", functionName, field, val);
+
+        if (!termFreqCache.containsKey(cacheKey)) {
+            Map<Object, Object> context = new HashMap<>() {
```

Comment:
No need to wrap to use context here, we could pass `indexSearcher` directly I believe